sql/database.go

```diff
@@ -654,19 +679,32 @@ type sqliteDatabase struct {
 
 	interceptMtx sync.Mutex
 	interceptors map[string]Interceptor
+
+	connWaitLatency prometheus.Observer
+	poolUsage       prometheus.Gauge
+
+	connIdleTimeout time.Duration
 }
 
 var _ Database = &sqliteDatabase{}
 
 func (db *sqliteDatabase) getConn(ctx context.Context) *sqlite.Conn {
 	start := time.Now()
 	conn := db.pool.Get(ctx)
-	if conn != nil {
-		connWaitLatency.Observe(time.Since(start).Seconds())
+	if conn != nil && db.connWaitLatency != nil {
+		db.connWaitLatency.Observe(time.Since(start).Seconds())
+		db.poolUsage.Inc()
 	}
 	return conn
 }
 
+func (db *sqliteDatabase) putConn(conn *sqlite.Conn) {
+	db.pool.Put(conn)
+	if db.poolUsage != nil {
+		db.poolUsage.Dec()
+	}
+}
+
```

Comment:
Instead of having to check those fields against `nil` how about setting a default metric name? The only thing this would impact is tests. There we don't read the metrics but collecting them might again affect timings or cause races that we won't detect if we are not using them.

Reply:
There's also `sql.Version()` which does not receive any config yet it uses an `sqliteDatabase`. I've added default database name but if we remove these checks we'll be getting unneeded `db` metrics entries from a single `sql.Version()` call early during startup.